OpenRA.Mods.Common/Activities/ExternalCaptureActor.cs

```diff
@@ -92,5 +90,20 @@ public override Activity Tick(Actor self)
 
 			return this;
 		}
+
+		void BeginCapture(Actor self)
+		{
+			capturable.BeginCapture(self);
+			if (conditionManager != null && !string.IsNullOrEmpty(capturesInfo.CapturingCondition) && capturingToken == ConditionManager.InvalidConditionToken)
+				capturingToken = conditionManager.GrantCondition(self, capturesInfo.CapturingCondition);
+		}
+
+		void EndCapture(Actor self)
+		{
+			if (capturable.CaptureInProgress)
```

Comment:
This needs to check `target.Type == TargetType.Actor` too, otherwise `capturable.EndCapture` will run on dead actors which then crashes the game with:
```
Exception of type `System.InvalidOperationException`: Attempted to get trait from destroyed object (proc 175 (not in world))
  at OpenRA.TraitDictionary.CheckDestroyed (OpenRA.Actor actor) [0x00026] in /Users/paul/src/OpenRA/OpenRA.Game//TraitDictionary.cs:84 
  at OpenRA.TraitDictionary.GetOrDefault[T] (OpenRA.Actor actor) [0x00002] in /Users/paul/src/OpenRA/OpenRA.Game//TraitDictionary.cs:94 
  at OpenRA.Actor.TraitOrDefault[T] () [0x0000d] in /Users/paul/src/OpenRA/OpenRA.Game//Actor.cs:248 
  at OpenRA.Mods.Common.Traits.ExternalCapturable.EndCapture () [0x00007] in /Users/paul/src/OpenRA/OpenRA.Mods.Common//Traits/ExternalCapturable.cs:80 
  at OpenRA.Mods.Common.Activities.ExternalCaptureActor.EndCapture (OpenRA.Actor self) [0x00017] in /Users/paul/src/OpenRA/OpenRA.Mods.Common//Activities/ExternalCaptureActor.cs:104 
  at OpenRA.Mods.Common.Activities.ExternalCaptureActor.Tick (OpenRA.Actor self) [0x00075] in /Users/paul/src/OpenRA/OpenRA.Mods.Common//Activities/ExternalCaptureActor.cs:41 
  at OpenRA.Activities.Activity.TickOuter (OpenRA.Actor self) [0x00061] in /Users/paul/src/OpenRA/OpenRA.Game//Activities/Activity.cs:175 
  at OpenRA.Traits.ActivityUtils.RunActivity (OpenRA.Actor self, OpenRA.Activities.Activity act) [0x00024] in /Users/paul/src/OpenRA/OpenRA.Game//Traits/ActivityUtils.cs:37 
  at OpenRA.Actor.Tick () [0x00010] in /Users/paul/src/OpenRA/OpenRA.Game//Actor.cs:163 
  at OpenRA.World.Tick () [0x000c3] in /Users/paul/src/OpenRA/OpenRA.Game//World.cs:345 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x00210] in /Users/paul/src/OpenRA/OpenRA.Game//Game.cs:604 
  at OpenRA.Game.LogicTick () [0x00055] in /Users/paul/src/OpenRA/OpenRA.Game//Game.cs:628 
  at OpenRA.Game.Loop () [0x000d6] in /Users/paul/src/OpenRA/OpenRA.Game//Game.cs:755 
  at OpenRA.Game.Run () [0x00042] in /Users/paul/src/OpenRA/OpenRA.Game//Game.cs:795 
  at OpenRA.Program.Run (System.String[] args) [0x00011] in /Users/paul/src/OpenRA/OpenRA.Game//Support/Program.cs:136 
  at OpenRA.Program.Main (System.String[] args) [0x00050] in /Users/paul/src/OpenRA/OpenRA.Game//Support/Program.cs:40 
```

Reply:
thanks, I added a test.
But how did you managed to crash the game ?
I tried to kill the captor and the target during the capture but nothing weird happened in both cases.

Reply:
For my tests I added `Inherits@foo: ^IronCurtainable` and `CapturingCondition: invulnerability` to `E6` in RA. I then tried out various situations ingame.  The crash occurs when the actor being captured (e.g. a power plant) is killed (e.g. using Tanya to demolish it) while the capture is underway.